app/gilded-rose.ts

```diff
@@ -19,51 +19,70 @@ export class GildedRose {
-            this.items[i].quality = this.items[i].quality - 1
-          }
-        }
-      } else {
-        if (this.items[i].quality < 50) {
-          this.items[i].quality = this.items[i].quality + 1
-          if (this.items[i].name == 'Backstage passes to a TAFKAL80ETC concert') {
-            if (this.items[i].sellIn < 11) {
-              if (this.items[i].quality < 50) {
-                this.items[i].quality = this.items[i].quality + 1
-              }
-            }
-            if (this.items[i].sellIn < 6) {
-              if (this.items[i].quality < 50) {
-                this.items[i].quality = this.items[i].quality + 1
-              }
-            }
-          }
-        }
+      if(!this.items[i].name.includes("Sulfuras",0)){
+        this.items[i].sellIn -= 1;
+        this.applyQualityChange(this.items[i], this.getItemQualityChange(this.items[i]));
       }
-      if (this.items[i].name != 'Sulfuras, Hand of Ragnaros') {
-        this.items[i].sellIn = this.items[i].sellIn - 1;
-      }
-      if (this.items[i].sellIn < 0) {
-        if (this.items[i].name != 'Aged Brie') {
-          if (this.items[i].name != 'Backstage passes to a TAFKAL80ETC concert') {
-            if (this.items[i].quality > 0) {
-              if (this.items[i].name != 'Sulfuras, Hand of Ragnaros') {
-                this.items[i].quality = this.items[i].quality - 1
-              }
-            }
-          } else {
-            this.items[i].quality = this.items[i].quality - this.items[i].quality
-          }
-        } else {
-          if (this.items[i].quality < 50) {
-            this.items[i].quality = this.items[i].quality + 1
-          }
-        }
+    }
+    return this.items;
+  }
+
+  getItemQualityChange(item:Item){
+    let change = this.getQualityChange(item);
+    if(item.name.includes("Backstage passes")) {
+      change = this.getPassesQualityChange(item);
+    }
+    else if(item.name === "Aged Brie"){
+      change = -change;
+    }
+
+    if(item.name.includes("Conjured")) {
+      change = 2*change;
+      if(item.name === "Conjured Aged Brie") {
+        change = -change;
       }
     }
+    return change;
+  }
 
-    return this.items;
+  getQualityChange(item:Item){
+    if(item.sellIn < 0) {
+      return -2;
+    }
+    return -1;
+  }
+
+  getPassesQualityChange(item:Item){
+    if(item.sellIn < 0) return -item.quality;
+    if(item.sellIn <= 5) {
+      return 3;
+    }
+    if(item.sellIn <= 10) {
+      return 2;
+    }
+    return 1;
+  }
+
+  applyQualityChange(item:Item, change:number) {
+    let newQuality = item.quality + change;
+    if(change < 0){
+      if(item.quality <= 0){
+        return;
+      }
+      if(newQuality <= 0) {
+        item.quality = 0;
+        return;
+      }
+    }
+    else if(change > 0){
+      if(item.quality >= 50){
+        return;
+      }
+      if(newQuality >= 50) {
+        item.quality = 50;
+        return;
+      }
+    }
+    item.quality = newQuality;
```

Comment:
I like that you're using `return` to exit from the function. One thing you can do here is to check if `change` is equal to 0, and exit the function earlier (at the top), so it's very clear that when the change is zero, nothing happens. This will also help flatten out your 1if/else blocks` a bit.